src/main/scala/apps/cameraPipeRewrite.scala

```diff
@@ -71,12 +71,12 @@ object cameraPipeRewrite {
     if (predicate(p)) { a(p) } else { b(p) }
   }
 
-  def isAppliedMap: Strategy[Rise] = function(function(isEqualTo(DSL.map)))
-  def isAppliedZip: Strategy[Rise] = function(function(isEqualTo(DSL.zip)))
-  def isAppliedDrop: Strategy[Rise] = function(depFunction(isEqualTo(DSL.drop)))
-  def isAppliedTake: Strategy[Rise] = function(depFunction(isEqualTo(DSL.take)))
+  def isAppliedMap: Strategy[Rise] = function(function(isEqualTo(p.map.primitive)))
```

Comment:
These could have remained with the `DSL.primitive` syntax.
As explained in another comment, I would like to see that we have only a single _DSL_ eventually.
This is one possibility of how this could then looks like (maybe a conversion from the TDSL type to the primitive could also be done implicitly, but I am not yet sure if that would be a good idea).

Reply:
Maybe we could use something like `untyped(map)` to avoid triggering type inference when not desired (I think type inference is triggered on TDSL -> Expr conversion otherwise, but this would bypass it, I need to double check the code).